contracts/credits/src/contract.rs

```diff
@@ -0,0 +1,1438 @@
+use ::cw20_base::ContractError as Cw20ContractError;
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, BankMsg, Binary, Coin, Deps, DepsMut, Env, MessageInfo, Response, StdError,
+    StdResult, Uint128,
+};
+use cw2::set_contract_version;
+use cw20::BalanceResponse;
+use cw20_base::state as Cw20State;
+use cw20_base::state::{BALANCES, TOKEN_INFO};
+
+use crate::msg::{
+    AllocationResponse, ConfigResponse, ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg,
+    TotalSupplyResponse, VestedAmountResponse, WithdrawableAmountResponse,
+};
+use crate::state::{Allocation, Config, Schedule, ALLOCATIONS, CONFIG};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:credits";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+const TOKEN_NAME: &str = "CNTRN";
+const TOKEN_SYMBOL: &str = "cuntrn";
```

Comment:
I think it should be just `cNTRN`

Reply:
Otherwise your `ucntrn` means that one `ucntrn` contains 10^6 some small tokens which is wrong.
The main token is  `cNTRN` which contains 10^6 `ucntrn` (and we don't define `ucntrn` in CW20)

Reply:
you're right, I misunderstood the meaning of this const